dali/pipeline/data/types.h

```diff
@@ -574,6 +578,7 @@ DALI_REGISTER_TYPE(string,         DALI_STRING);
 DALI_REGISTER_TYPE(DALIImageType,  DALI_IMAGE_TYPE);
 DALI_REGISTER_TYPE(DALIDataType,   DALI_DATA_TYPE);
 DALI_REGISTER_TYPE(DALIInterpType, DALI_INTERP_TYPE);
+DALI_REGISTER_TYPE(DALIColorFilter, DALI_COLOR_FILTER);
```

Comment:
Do we really, _really_ need this? I don't like adding stuff to types.h and, honestly, I'd very much love to _remove_ other enums (like ImageType and InterpType) from here instead of adding more.

Reply:
I guess it is best we have for now. We need the ability to specify the one of four avialable options per-sample as tensor inputs. I've tried going around it with np.bytes_("bg") in external source, but 1. it requires "parsing" strings per-sample in each iteration, 2. is unfamiliar to users, 3. has limitations, for example, I don't see a way to create a tensor of strings that could be digested by DALI.

Reply:
If the motivation for this is to allow the user to specify the pattern per sample, then the likeliest reason is so that you can demosaic a crop. In this case, dealing with blue (or red) position within a Bayer cell would be much easier to work with:
```
(blue coordinates in XY order)
color_filter=(0,0)  # BGGR
color_filter=(0,1)  # GRBG
color_filter=(1,0) # GBRG
color_filter=(1,1) # RGGB 
```
If you know your original layout, you can calculate the layout of the crop as:
`(layout_x - crop_x)&1, (layout_y - crop_y)&1`

Reply:
As per slack disussion: Reverting the exposure of enum, because passing the enums per-sample is cumbersome (requires reinterpret op along the way),